src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,146 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int START_CAPACITY = 16;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = new Node[START_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
```

Comment:
Why do you need this IF. It seems you can delete  `putForNullKey` - I can not see any reason for it.